api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto

```diff
@@ -301,4 +302,10 @@ message ExtProcOverrides {
   // config used. It is the prerogative of the control plane to ensure this
   // most-specific config contains the correct final overrides.
   MetadataOptions metadata_options = 6;
+
+  // Additional metadata to include into streams initiated to the ext_proc gRPC
+  // service. This can be used for scenarios in which additional ad hoc
+  // authorization headers (e.g. ``x-foo-bar: baz-key``) are to be injected or
+  // when a route needs to partially override inherited metadata.
+  repeated config.core.v3.HeaderValue grpc_metadata = 7;
```

Comment:
This PR is fine from an API perspective, will defer to others to merge. One slight nit though, I probably would make this `grpc_initial_metadata` if it was me given all the other metadata options and contexts floating around.

Reply:
👍  - renamed to `grpc_initial_metadata`